administration/src/bp-modules/applications/ApplicationsOverview.tsx

```diff
@@ -178,13 +178,28 @@ export class ApplicationViewComponent extends React.Component<{
   }
 }
 
+const sortByStatus = (a: number, b: number): number => a - b
+const sortByDateAsc = (a: Date, b: Date): number => a.getTime() - b.getTime()
+
+const getVerificationStatus = (status: number[]): VerificationStatus => {
+  if (status.every(val => val === VerificationStatus.Verified)) return VerificationStatus.Verified
+  if (status.every(val => val === VerificationStatus.Rejected)) return VerificationStatus.Rejected
+  return VerificationStatus.Awaiting
+}
+
+// Applications will be sorted by unique status which means fully verified/rejected and within this status by creation date asc
+const sortApplications = (applications: Application[]): Application[] =>
+  applications
+    .map(application => ({ ...application, status: getVerificationStatus(application.verifications.map(getStatus)) }))
+    .sort((a, b) => sortByStatus(a.status, b.status) || sortByDateAsc(new Date(a.createdDate), new Date(b.createdDate)))
+
 const ApplicationsOverview = (props: { applications: Application[] }) => {
   const [updatedApplications, setUpdatedApplications] = useState(props.applications)
   const { applicationIdForPrint, printApplicationById } = usePrintApplication()
 
```

Comment:
Maybe you can use `useMemo` to create the sorted array only when the set of applications changes, and not every time this component is rendered. Something like this:

```ts
    const sortedApplications = useMemo(() => sortApplications(applications), [updatedApplications])
```

Reply:
ah okay i thought if i define the function outside of the component it will not be rerendered every time. Where i defined the function there is no `props.applications` i think these hook are useful inside the component when they depend on state and/or lifecycle hooks and you can not define them outside @michael-markl . But maybe i got that wrong

Reply:
React does (and can) not know if the sorting happens within an external function or not. Any call to an external function will be done whenever the component is rendered (ie. when the function `ApplicationsOverview` is exectued).